.github/scripts/hermetic_library_generation.sh

```diff
@@ -75,8 +75,16 @@ workspace_name="/workspace"
 baseline_generation_config="baseline_generation_config.yaml"
 message="chore: generate libraries at $(date)"
 
+# there's no local branch when running in GitHub action,
+# thus no need to use `git pull`.
 git checkout "${target_branch}"
 git checkout "${current_branch}"
+# only allow fast-forward merging; exit with non-zero result if there's merging
+# conflict.
+if [[ $(git merge -m "chore: merge ${target_branch} into ${current_branch}" "${target_branch}") -ne 0 ]]; then
```

Comment:
Another way is to always update the branch to main before updating in https://github.com/googleapis/sdk-platform-java/blob/main/.github/scripts/update_googleapis_commit.sh, so that we don't have to deal with potential conflict.

Reply:
Hmm, maybe change in `update_googleapis_commit.sh` is better because we only want to merge main in generation PR and other pull requests need manual merge anyway.